### PR TITLE
Add CSMSC reicpe

### PR DESCRIPTION
**Pull request by kan-bayashi**
This PR adds CSMSC recipe (Chinese, single female speaker).
https://www.data-baker.com/open_source.html
(Should be merged after #1258)

- [x] make recipes
- [x] update egs/README
- [x] upload pretrained model and update RESULTS.md

### CODE REVIEW

**Comment by kan-bayashi** [comment]
Almost done.

**Comment by kan-bayashi** [comment]
This is an example of json.

```json
        "000201": {
            "input": [
                {
                    "feat": "/work5/t_hayashi/work/espnet_dev/egs/csmsc/tts1/dump/train_no_dev/feats.1.ark:7",
                    "name": "input1",
                    "shape": [
                        253,
                        80
                    ]
                }
            ],
            "output": [
                {
                    "name": "target1",
                    "shape": [
                        19,
                        261
                    ],
                    "text": "l iang3 s ou1 ch uan2 z ai4 h u4 h ang2 uang3 x iang1 g ang2 z ou3",
                    "token": "l iang3 s ou1 ch uan2 z ai4 h u4 h ang2 uang3 x iang1 g ang2 z ou3",
                    "tokenid": "151 89 176 166 39 197 258 10 71 183 71 20 203 257 87 70 20 258 168"
                }
            ],
            "utt2spk": "csmsc"
        },
        "000202": {
            "input": [
                {
                    "feat": "/work5/t_hayashi/work/espnet_dev/egs/csmsc/tts1/dump/train_no_dev/feats.1.ark:20915",
                    "name": "input1",
                    "shape": [
                        198,
                        80
                    ]
                }
            ],
            "output": [
                {
                    "name": "target1",
                    "shape": [
                        15,
                        261
                    ],
                    "text": "er2 z eng1 m u3 ie3 h uan4 iou3 t ang2 n iao4 b ing4",
                    "token": "er2 z eng1 m u3 ie3 h uan4 iou3 t ang2 n iao4 b ing4",
                    "tokenid": "65 258 56 152 182 105 71 199 140 179 20 153 99 37 127"
                }
            ],
            "utt2spk": "csmsc"
        },
```

**Comment by kan-bayashi** [comment]
I generated speech with Transformer @ 200 epochs.
The quality sounds very nice (but I'm not sure the pronunciation is correct).
https://drive.google.com/open?id=1Si5CJ-b3iLKjW5bs-0Ll2EBnoCqMO_3n

File: egs/csmsc/tts1/run.sh [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+resume=""    # the snapshot path to resume (if set empty, no effect)
+
+# feature extraction related
+fs=24000      # sampling frequency
+fmax=7600     # maximum frequency
+fmin=80       # minimum frequency
+n_mels=80     # number of mel basis
+n_fft=1024    # number of fft points
+n_shift=256   # number of shift points
+win_length="" # window length
```
Comment (r9y9) [inline egs/csmsc/tts1/run.sh]: Would it be possible to change these parameters to n_fft=2048, n_shift=300, win_length=1200? (12.5ms shift, and 50ms frame length) 

These seem to be common in recent E2E TTS research using 24kHz dataset.

Comment (kan-bayashi) [reply egs/csmsc/tts1/run.sh]: I want to unify hyperparameters, but it is OK.
At least we should use the same setting for the same sampling rate.

Comment (r9y9) [reply egs/csmsc/tts1/run.sh]: For 22.5k (e.g. LJSpeech):
- n_fft=1024, n_shift=256

For 24k
- n_fft=2048, n_shift=300, win_length=300  (same as Tacotron 2 paper; 12.5ms shift, 50 ms frame len)

Does this sound good for you?

Comment (kan-bayashi) [reply egs/csmsc/tts1/run.sh]: OK but I’ve never compare on 24khz, so I’m not sure this difference makes any improvement.
Do you have some experience?

Comment (r9y9) [reply egs/csmsc/tts1/run.sh]: Well, I’m just following a (sort of) community standard. I guess there wouldn’t be significant difference, but if we use the same parameters with the existing method, results will be more comparable. So, if we support tacotron 2, why not using same parameters as tacotron 2?

In my limited experience, there’s no huge difference. As I mentioned in #795, training would be faster if we use larger shift.

Comment (kan-bayashi) [reply egs/csmsc/tts1/run.sh]: OK. I changed my mind.
From now, we should keep 

For 22.5k (e.g. LJSpeech):

n_fft=1024, n_shift=256
For 24k

n_fft=2048, n_shift=300, win_length=300 (same as Tacotron 2 paper; 12.5ms shift, 50 ms frame len)

**Comment by kan-bayashi** [comment]
I started the training of new feature configuration.
ETA: Tomorrow evening

**Comment by kan-bayashi** [comment]
Finished. Pretrained model, feature and samples are available in google drive.
I think the quality is quite nice.

```
# v.0.5.3 Initial Transformer

- FTT in points: 2048
- Shift in points: 256
- Window in points: 1200
- Frequency limit: 80-7600
- Fast-GL 64 iters

## Environments
- date: `Mon Oct 14 09:59:15 JST 2019`
- python version: `3.7.3 (default, Mar 27 2019, 22:11:17)  [GCC 7.3.0]`
- espnet version: `espnet 0.5.3`
- chainer version: `chainer 6.0.0`
- pytorch version: `pytorch 1.0.1.post2`
- Git hash: `36321a262b8dbb6c13297d20e050192624366517`
  - Commit date: `Sat Oct 12 13:24:10 2019 +0900`

## Models

- model link: https://drive.google.com/open?id=1bTSygvonv5TS6-iuYsOIUWpN2atGnyhZ
- training config file: `conf/tuning/train_pytorch_transformer.v1.single.yaml`
- decoding config file: `conf/decode.yaml`
- cmvn file: `data/train_no_dev/cmvn.ark`
- e2e file: `exp/train_no_dev_pytorch_train_pytorch_transformer.v1.single/results/model.last1.avg.best`
- e2e JSON file: `exp/train_no_dev_pytorch_train_pytorch_transformer.v1.single/results/model.json`
- dict file: `data/lang_phn/train_no_dev_units.txt`

## Samples

https://drive.google.com/open?id=1OgDwp8GVInnqda5ohE_U2DXXrHFFIa17
```

**Comment by kan-bayashi** [comment]
Thanks!

**Pull request merged**